File: src/Pascal's Triangle.cpp
#include <string>
#include <vector>
#include <iostream>
#include <set>
#include <map>
#include <algorithm>
#include <queue>
#include <limits.h>
#include <stdio.h>
#include <string.h>
using namespace std;

class Solution {
public:
    vector<vector<int> > generate(int numRows) {
        vector<vector<int> > ret;
        for(int i = 0; i < numRows; i++)
        {
            vector<int> e;
            for(int j = 0; j <= i; j++)
            {
                if(j == 0 || j == i)
                    e.push_back(1);
                else
                    e.push_back(ret[i-1][j-1] + ret[i-1][j]);
            }
            ret.push_back(e);
        }
        return ret;
    }
};

int main()
{
    Solution so;
    so.generate(1);
    return 0;
}

File: python/Pascal's Triangle.py
class Solution:
    # @return a list of lists of integers
    def generate(self, numRows):
        ret = []
        for i in range(numRows):
            l = []
            for j in range(i+1):
                if j == 0 or j == i:
                    l.append(1)
                else:
                    l.append(ret[i-1][j-1] + ret[i-1][j])
            ret.append(l)
        return ret